osbuild/formats/v1.py

```diff
@@ -32,7 +32,7 @@ def describe_pipeline(pipeline: Pipeline) -> Dict[str, Any]:
             build = manifest[pipeline.build]
             description["build"] = {
                 "pipeline": describe_pipeline(build),
-                "runner": pipeline.runner
+                "runner": pipeline.runner.name
```

Comment:
Shouldn't we retain the original value here? You now make `osbuild --describe` dependent on the local host, since it will now produce different values depending on which runner-symlinks/entries are on your host? (Same in v2)

You could just retain the original value in `Runner` as `original_name` or something like that? Or am I reading this wrong?

Reply:
`Runner.name` is the original name given in the manifest, e.g. the v2 loading code:

```
     runner_name = description.get("runner")
     runner = None
     if runner_name:
         runner = Runner(index.detect_runner(runner_name), runner_name)
     else:
         runner = Runner(index.detect_host_runner())
```
and
```
class Runner:
     def __init__(self, info, name: Optional[str] = None) -> None:
         self.info = info  # `meta.RunnerInfo`
         self.name = name or os.path.basename(info.path)
```

Reply:
In both formats we only enter that code path if conditionally on the pipeline having a build pipeline, i.e. `pipeline.build` is set. Or the other way around, if there is no build pipeline set for a pipeline, it will use the host runner, but then we don't actually emit the runner.
This is all a bit convoluted because the internal representation does not match the manifest json, i.e. in the json we specify the runner that the pipeline should be executed with but in our data model`Pipeline.runner` it is the runner for `Pipeline.build`. I think we should clean this up, and have `Pipeline.build` refer to the actual `Pipeline` object and then we match the representation (This would also take care of the mypy situation around `Pipeline.runner`).